src/cmap/background.rs

```diff
@@ -24,25 +27,24 @@ pub(crate) fn start_background_task(pool: Weak<ConnectionPoolInner>) {
 async fn perform_checks(pool: ConnectionPool) {
     // We remove the perished connections first to ensure that the number of connections does not
     // dip under the min pool size due to the removals.
-    pool.remove_perished_connections().await;
-    pool.ensure_min_connections().await;
+    pool.inner.remove_perished_connections().await;
+    pool.inner.ensure_min_connections().await;
 }
 
-impl ConnectionPool {
+impl ConnectionPoolInner {
     /// Iterate over the connections and remove any that are stale or idle.
     async fn remove_perished_connections(&self) {
-        let mut connection_manager = self.inner.connection_manager.lock().await;
+        let mut available_connections = self.available_connections.lock().await;
```

Comment:
I imagine this process will be really quick, so I just left it as holding the lock for the duration.

Reply:
Good idea! Updated to quit early when encountering a non-idle, non-stale connection. Also updated to release the lock between iterations.